uml_lib/ebAPI_lib.py

```
@@ -67,58 +71,262 @@
+        # Open the URL with timeout and decode using server-provided charset (default utf-8).
+        with urlopen(req, timeout=timeout) as resp:
+            charset = resp.headers.get_content_charset() or "utf-8"
+            return resp.read().decode(charset)
+    except HTTPError as e:
+        # If the server returns an error status, try to include its body in the exception.
+        try:
+            body = e.read().decode("utf-8", errors="replace")
+        except Exception:
+            body = ""
+        # Re-raise HTTPError with an augmented message that includes the body preview.
+        raise HTTPError(e.url, e.code, f"{e.reason}. Response body: {body}", e.headers, e.fp)
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)